app/controllers/providers/means/vehicle_details_controller.rb

```diff
@@ -13,7 +17,13 @@ def update
     private
 
       def vehicle
-        @vehicle ||= legal_aid_application.vehicle || legal_aid_application.build_vehicle
+        @vehicle ||= existing_vehicle || legal_aid_application.vehicles.build
+      end
+
+      def existing_vehicle
+        legal_aid_application.vehicles.find(params["id"])
+      rescue ActiveRecord::RecordNotFound
+        nil
```

Comment:
nit, I know we use this pattern in a fair few places but it can be simplified

```
Vehicle.find "asdd"
=> Couldn't find Vehicle with 'id'=asdd (ActiveRecord::RecordNotFound)
Vehicle.find_by id: "asdd"
=> nil

Reply:
I thought that too, but I'm sure I had to re-add the rescue block because the controller was failing... I'll investigate :+1:

Reply:
Yeah, I think it's because it's called from the memo'd method above

```ruby
      def vehicle
        @vehicle ||= existing_vehicle || legal_aid_application.vehicles.build
      end

      def existing_vehicle
        legal_aid_application.vehicles.find(params["id"])
      rescue ActiveRecord::RecordNotFound
        nil
      end
```

```
@vehicle ||= nil || legal_aid_application.vehicles.build
```
and

```
@vehicle ||= raise ActiveRecord::RecordNotFound || legal_aid_application.vehicles.build
```

Work okay, but calling the method without a rescue fails :/